pennylane/devices/qubit/apply_operation.py

```diff
@@ -238,10 +238,12 @@ def apply_operation_csr_matrix(op, state, is_state_batched: bool = False):
     """The csr_matrix specialized version apply operation."""
     # State is numpy array, should have been stored in tensor version
     # remember the initial shape and recover in the end
+    if sp.sparse.issparse(state):
+        raise TypeError("State should not be sparse in default qubit pipeline")
```

Comment:
How would we get a sparse state in the first place?

Reply:
It won't happen for the whole pipeline and any common users. 
This is to block such misuse in our own dev. One previous test fail was caused by such reason and the error was not clear because it only emerges in deeper callstack. Having such blocker would help prevent such scenario.